src/sentry/notifications/utils/tasks.py

```diff
@@ -1,21 +1,51 @@
 from __future__ import annotations
 
 from collections.abc import Iterable, Mapping
+from datetime import datetime
 from typing import TYPE_CHECKING, Any
 
 from django.apps import apps
+from django.utils.functional import SimpleLazyObject
 
 from sentry.db.models import Model
 from sentry.notifications.class_manager import NotificationClassNotSetException, get
 from sentry.silo.base import SiloMode, region_silo_function
 from sentry.tasks.base import instrumented_task
 from sentry.taskworker.config import TaskworkerConfig
 from sentry.taskworker.namespaces import notifications_tasks
+from sentry.users.services.user.model import RpcUser
 
 if TYPE_CHECKING:
     from sentry.notifications.notifications.base import BaseNotification
 
 
+def serialize_lazy_object(arg: SimpleLazyObject, key: str | None = None) -> dict[str, Any]:
+    raw_data = arg.dict()
+    parsed_data = {}
+    for k, v in raw_data.items():
+        if isinstance(v, datetime):
+            v = v.isoformat()
+
+        parsed_data[k] = v
+
+    return {
+        "type": "lazyobject",
```

Comment:
We'll also need the type of the object inside `SimpleLazyObject`. I think it will most likely be `User` or `RpcUser`.